src/main/cli/settings.c

```diff
@@ -1656,11 +1656,11 @@ const clivalue_t valueTable[] = {
 #endif
 
 #ifdef USE_RPM_FILTER
-    { "gyro_rpm_notch_harmonics",  VAR_UINT8 | MASTER_VALUE, .config.minmaxUnsigned = { 0, 3 }, PG_RPM_FILTER_CONFIG, offsetof(rpmFilterConfig_t, gyro_rpm_notch_harmonics) },
```

Comment:
These parameter names were introduced in Betaflight 4.0, and have been around for over 3 years. They feature in countless forum posts, videos, snippets that people use. Changing them will cause a lot of disruption, for no real value, as the CLI is not meant as a UI for users to go and 'find things'.

Reply:
The one thing I feel strongly about is that `gyro_rpm_notch_min` needs to have 'hz' appended to the end of it.

Otherwise just `gyro_rpm_notch_min` is confusing.  On the face of it, we should enter the minimum rpm value below which the filter cannot go, in the same way that 'dyn_idle_min_rpm` requires an RPM input (in the form 'rpm/100').  

As it happens, `gyro_rpm_notch_min` expects an input in 'hz/100', not 'rpm/100'.  If someone enters 40, they will get a filter that works down to 2400 RPM, not the 4000 RPM they (probably) intended.  

Hence we should definitely re-name `gyro_rpm_notch_min` to `gyro_rpm_notch_min_hz`, to make explicit what we want to be entered here.

The more general name change proposed here is to remove `gyro_` from the start, and to make each element start with the unique and identical string `rpm_filter_`.  This corrects the inconsistent naming of the parameters that we now have.  

You'll recall that originally the RPM filter was provided in gyro_ and dterm_ options.  That's why the four non-common parameters originally had `gyro_` pre-pended - to distinguish them from the `dterm` versions of the same filter.  Starting in 4.2 the `dterm_` options were removed;  and since then, the RPM filter applies only to gyro.  But the name still carries over `gyro_` in four of the five parameters.

When auto-completing an rpm filter related CLI value, we will only reliably get all relevant parameters if the initial characters are unique to what we are looking for.  With this PR, we only have to type 'rpm_f' as the initial string, and we get exactly he 5 parameters we want.  Otherwise It's frustrating to have to remember to prepend 'gyro_', when you're thinking `rpm_` - and if you do search `gyro_` you get a lot of junk, and you won't pick up 'rpm_notch_hz', which doesn't have `gyro_` before it.

Hence it will be a better user experience if we were to remove `gyro_` from the start of these RPM filter parameters, and make them all start with `rpm_filter_`.

Since we will be using CLI snippet based presets in the future, it is best to fix these naming idiosyncrasies before they are baked into presets  It will be far more difficult to change the names late onr.  

I agree that there's no strong need to change the blackbox log header names, however it is not difficult to make a version specific change to adapt to the new names, and I will make a PR for that.  Or we can leave them as is.  However if we make all the names in the code also consistently start with  `rpm_filter`, a global search for the rpm filter code will work more reliably.

Lastly it's likely that we will change the defaults for these a bit from 4.2 to 4.3,  A name change will push the new defaults with them, and that isn't a bad thing, I think.


Hence I think we should make these changes now, that they will be useful to most users.

Reply:
I strongly support @ctzsnooze points concerning appending "_hz" to `gyro_rpm_notch_min`** to make the syntax consistent.
=> `gyro_rpm_notch_min_hz`

**ctzsnooze wrote:**

> As it happens, gyro_rpm_notch_min expects an input in 'hz/100', not 'rpm/100'. If someone enters 40, they will get a filter that works down to 2400 RPM, not the 4000 RPM they (probably) intended.

`gyro_rpm_notch_min` expects an input in normal 'hz', not 'hz/100'. So if you enter a 40, the RPM notchhes will work down to 40Hz (40Hz * 60 = 2400RPM).

That's why I'm very much for appending '_hz' to `gyro_rpm_notch_min` so that misunderstandings like these (Hz vs. RPM) don't happen.

-----

I can also understand @mikeller's reasons for not changing the other CLI commands, but since we are already changing a lot of CLI commands, I guess it's a good time to clean the naming up right now.